distutils/extension.py

```diff
@@ -105,9 +105,9 @@ def __init__(
         **kw,  # To catch unknown keywords
     ):
         if not isinstance(name, str):
-            raise AssertionError("'name' must be a string")
+            raise TypeError("'name' must be a string")
```

Comment:
This is a breaking change.  However, I understand the `AssertionError` was supposed to be a catastrophic event any way. Raising a different exception is also a catastrophic event. It is slightly different, but then a catastrophe is a catastrophe.

Reply:
As you can see from other cases that changed in this diff, there were places that AssertionError was explicitly caught. I think your instinct is right here, but it could cause breakage.

I suspect these were originally `assert` statements that were converted to `raise AssertionError` for correctness under python -O (which no one uses).

Reply:
Agreed.